cmd/csi-addons/replication.go

```diff
@@ -0,0 +1,245 @@
+	k := getKubernetesClient()
+
+	rs := service.NewReplicationServer(rep.Client, k)
+
+	req := &proto.EnableVolumeReplicationRequest{
+		SecretName:      rep.secretName,
+		SecretNamespace: rep.secretNamespace,
+		VolumeId:        rep.volumeID,
+	}
+
+	_, err := rs.EnableVolumeReplication(context.TODO(), req)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("Enable Volume Replication operation successful")
+
+	return nil
+}
+
+// DisableVolumeReplication executes the DisableVolumeReplication operation.
+type DisableVolumeReplication struct {
+	VolumeReplicationBase
+}
+
+var _ = registerOperation("DisableVolumeReplication", &DisableVolumeReplication{})
+
+func (rep *DisableVolumeReplication) Execute() error {
+	k := getKubernetesClient()
+
+	rs := service.NewReplicationServer(rep.Client, k)
+
+	req := &proto.DisableVolumeReplicationRequest{
+		SecretName:      rep.secretName,
+		SecretNamespace: rep.secretNamespace,
+		VolumeId:        rep.volumeID,
+	}
+
+	_, err := rs.DisableVolumeReplication(context.TODO(), req)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("Disable Volume Replication operation successful")
+
+	return nil
+}
+
+// PromoteVolume executes the PromoteVolume operation.
+type PromoteVolume struct {
+	VolumeReplicationBase
+}
+
+var _ = registerOperation("PromoteVolume", &PromoteVolume{})
+
+func (rep *PromoteVolume) Execute() error {
+	k := getKubernetesClient()
+
+	rs := service.NewReplicationServer(rep.Client, k)
+
+	req := &proto.PromoteVolumeRequest{
+		SecretName:      rep.secretName,
+		SecretNamespace: rep.secretNamespace,
+		VolumeId:        rep.volumeID,
+	}
+
+	_, err := rs.PromoteVolume(context.TODO(), req)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("Promote Volume operation successful")
+
+	return nil
+}
+
+// DemoteVolume executes the DemoteVolume operation.
+type DemoteVolume struct {
+	VolumeReplicationBase
+}
+
+var _ = registerOperation("DemoteVolume", &DemoteVolume{})
+
+func (rep *DemoteVolume) Execute() error {
+	k := getKubernetesClient()
+
+	rs := service.NewReplicationServer(rep.Client, k)
+
+	req := &proto.DemoteVolumeRequest{
+		SecretName:      rep.secretName,
+		SecretNamespace: rep.secretNamespace,
+		VolumeId:        rep.volumeID,
+	}
+
+	_, err := rs.DemoteVolume(context.TODO(), req)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("Demote Volume operation successful")
+
+	return nil
+}
+
+// ResyncVolume executes the ResyncVolume operation.
+type ResyncVolume struct {
+	VolumeReplicationBase
+}
+
+var _ = registerOperation("ResyncVolume", &ResyncVolume{})
+
+func (rep *ResyncVolume) Execute() error {
+	k := getKubernetesClient()
+
+	rs := service.NewReplicationServer(rep.Client, k)
+
+	req := &proto.ResyncVolumeRequest{
+		SecretName:      rep.secretName,
+		SecretNamespace: rep.secretNamespace,
+		VolumeId:        rep.volumeID,
+	}
+
+	_, err := rs.ResyncVolume(context.TODO(), req)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("Resync Volume operation successful")
```

Comment:
instead of Printf use Println which adds a new line, take care of it in all other places as well

Reply:
Thanks for the review Madhu, Updated the PR.
PTAL